core/src/main/java/org/infinispan/reactive/publisher/impl/LocalPublisherManagerImpl.java

```diff
@@ -362,8 +527,8 @@ public void segmentsLost(IntSet lostSegments) {
             return Flowable.just(value);
          }
          return RxJavaInterop.<R>completionStageToPublisher().apply(stage);
-      });
-      return finalizer.apply(resultPublisher);
+         // Have to make sure that flatMap parallelism is at least the # of segments - otherwise we can starve groupBy
+      }, maxSegment);
```

Comment:
Where's the `groupBy`?

Reply:
Forgot I rewrote the nonsegmented one to no longer use group, this can be removed.